antfs-cli and openant: init at unstable-2017-02-11

richardlarocque opened the pull request:
###### Motivation for this change

Adds `antfs-cli` and the `openant` library it depends on to nixpkgs.

These packages seem to not have had any formal release and their version numbers haven't been updated in a long time, so I decided to package them as a Git snapshot rather than aiming for a particular (non-existent) release.

###### Things done

- [ ] Tested using sandboxing
  ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS,
    or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file)
    on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] Linux
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

richardlarocque committed:
openant: init at unstable-2017-02-11

richardlarocque committed:
antfs-cli: init at unstable-2017-02-11

rycee commented on pkgs/top-level/python-packages.nix:
```diff
@@ -169,6 +169,26 @@ in {
     };
   };
 
+  antfs-cli = buildPythonApplication rec {
```
Since this is an application, it should go into the directory structure similarly to garmin-uploader.

rycee commented on pkgs/top-level/python-packages.nix:
```diff
@@ -169,6 +169,26 @@ in {
 
+  antfs-cli = buildPythonApplication rec {
+    name = "antfs-cli-${version}";
+    version = "git-2017-02-11";
```
Since the version field is not actually used anywhere in the expression you can inline it into the name field. Note, however, that this will not be interpreted as you might expect. Nix will parse name as `antfs-cli-git` and the version to `2017-02-11`. This is because it starts the version at the first hyphen followed by a digit:
```
nix-repl> builtins.parseDrvName "antfs-cli-git-2017-02-11"
{ name = "antfs-cli-git"; version = "2017-02-11"; }
```

Also as per the [naming guidelines](https://nixos.org/nixpkgs/manual/#sec-package-naming) it should be `name = "antfs-cli-unstable-2017-02-11";`.

This comment also applies for the openant package below.

richardlarocque commented:
Thanks for the quick review!  I'm not so quick myself, so feel free to take your time on the second round.

I've fixed a few things and squashed my fixes.  Changes include:
- Renaming and re-versioning both packages as "-unstable-2017-02-11".
- Add `platforms = platform.linux` to meta attributes.  (I presume this package isn't intended to work elsewhere as it has install scripts that are hard-coded to install udev rules.)  
- Moved antfs-cli to `pkgs/applications/misc`

FRidh commented on pkgs/top-level/python-packages.nix:
```diff
@@ -4659,6 +4659,37 @@ in {
+
+    # Removes some setup.py hacks intended to install udev rules.
+    # We do the job ourselves in postInstall below.
+    patchPhase = ''
```
make it `postPatch` since that allows us to still pass in patches`

rycee commented:
Beside the @FRidh's point about `postPatch` this looks fine to me. I don't have my Garmin watch around at the moment but hope to use this package in the future :-)

richardlarocque commented:
Replaced `patchPhase` with `postPatch`.  Also updated the PR description, which I just noticed was out of date.
